packages/svelte/src/preprocessors.ts

```diff
@@ -0,0 +1,66 @@
+import MagicString from 'magic-string';
+
+import { ComponentTrackingInitOptions, PreprocessorGroup, TrackComponentOptions } from './types';
+
+const defaultComponentTrackingOptions: Required<ComponentTrackingInitOptions> = {
+  trackComponents: true,
+  trackMount: true,
+  trackUpdates: true,
+};
+
+/**
+ * Svelte Preprocessor to inject Sentry performance monitoring related code
+ * into Svelte components.
+ */
+export function componentTrackingPreprocessor(options?: ComponentTrackingInitOptions): PreprocessorGroup {
+  const mergedOptions = { ...defaultComponentTrackingOptions, ...options };
+
+  return {
+    // This script hook is called whenever a Svelte component's <script>
+    // content is preprocessed.
+    // `content` contains the script code as a string
+    script: ({ content, filename }) => {
+      if (!shouldInjectFunction(mergedOptions.trackComponents, filename)) {
+        return { code: content };
+      }
+
+      const { trackMount, trackUpdates } = mergedOptions;
+      const trackComponentOptions: TrackComponentOptions = {
+        trackMount,
+        trackUpdates,
+        componentName: getComponentName(filename || ''),
+      };
+
+      const importStmt = 'import { trackComponent } from "@sentry/svelte";\n';
+      const functionCall = `trackComponent(${JSON.stringify(trackComponentOptions)});\n`;
+
+      const s = new MagicString(content);
+      s.prepend(functionCall).prepend(importStmt);
+
+      const updatedCode = s.toString();
+      const updatedSourceMap = s.generateMap().toString();
+
+      return { code: updatedCode, map: updatedSourceMap };
+    },
+  };
+}
+
+function shouldInjectFunction(
+  trackComponents: Required<ComponentTrackingInitOptions['trackComponents']>,
+  filename: string | undefined,
+): boolean {
+  if (!trackComponents || !filename) {
+    return false;
+  }
+  if (Array.isArray(trackComponents)) {
+    // TODO: this probably needs to be a little more robust
+    const componentName = getComponentName(filename);
+    return trackComponents.some(allowed => allowed === componentName);
+  }
+  return true;
+}
+function getComponentName(filename: string): string {
```

Comment:
There is a node api function that does exactly what this function is doing: `path.basename(filename, '.svelte')`. No need to change this though.

Reply:
So I tried replacing this with `path` but my Svelte app won't build because I'm using a Node API. There's probably a way to get around this but for now, I'll leave it as is. We can revisit this, if we ever get problems with that function

Reply:
IMO we can just rename `getComponentName` to reflect what it's doing (getting the basename) and not worry about using the path API.